### PR TITLE
Dark mode: Review detail

### LINKED ISSUE #1885: Dark mode: Review detail
Implement light/dark mode on the review detail screen.

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #1885. 

I've kept the original toggle button for now.

image

image

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/res/layout/fragment_review_detail.xml [hunk 1]
```diff
@@ -149,38 +145,36 @@
...
             <ToggleButton
                 android:id="@+id/review_approve"
+                style="@style/Woo.Button.Toggle"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
-                android:layout_marginBottom="@dimen/margin_medium"
-                android:layout_marginEnd="@dimen/margin_extra_large"
+                android:layout_marginBottom="@dimen/minor_100"
+                android:layout_marginEnd="@dimen/major_100"
                 android:background="@android:color/transparent"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/fragment_review_detail.xml]: It's so strange that there isn't a `MaterialToggleButton` (which is why some of the `Woo.Button` style settings aren't transferring properly). Recommend moving this `android:background="@android:color/transparent"` to inside the new `Woo.Button.Toggle` style for now. Hopefully someday they create a Material option.

File: WooCommerce/src/main/res/layout/fragment_review_detail.xml [hunk 2]
```diff
@@ -12,135 +12,131 @@
...
 
                     <ImageView
                         android:id="@+id/review_product_icon"
-                        android:layout_width="@dimen/product_icon_sz"
-                        android:layout_height="@dimen/product_icon_sz"
+                        android:layout_width="@dimen/image_minor_100"
+                        android:layout_height="@dimen/image_minor_100"
                         android:importantForAccessibility="no"
                         android:scaleType="centerCrop"
                         android:src="@drawable/ic_product"/>
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/fragment_review_detail.xml]: `ic_product.xml` is using `wc_border_color` (#E6E6E6) and likely needs to be updated for light/dark mode : 

Screen Shot 2020-02-04 at 8 16 06 PM

@Garance91540 WDYT?

File: WooCommerce/src/main/res/values/styles_base.xml [hunk 3]
```diff
@@ -80,6 +80,7 @@ theme across the entire app. Overridden versions should be added to the styles.x
         Card Styles
     -->
     <style name="Woo.Card" parent="@style/Widget.MaterialComponents.CardView">
+        <item name="android:background">@color/card_bgColor</item>
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/values/styles_base.xml]: This should be removed. This style should only be used with `MaterialCardView` which will take care of setting elevation and the appropriate alpha.

File: WooCommerce/src/main/res/values/colors.xml [hunk 4]
```diff
@@ -58,7 +58,7 @@
     <color name="menu_action_text">#FFFFFFFF</color>
 
     <!-- Default Card colors -->
-    <color name="card_bgColor">@color/white</color>
+    <color name="card_bgColor">@color/color_surface</color>
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/values/colors.xml]: Colors in the `colors.xml` file should not be edited. These are all old styles. Any new colors needed should be added to `colors_base.xml`. Either way this change is not needed - see code comments in the `styles_base.xml` file.

File: WooCommerce/src/main/res/layout/fragment_review_detail.xml [hunk 5]
```diff
@@ -12,135 +12,131 @@
...
-                android:layout_marginStart="@dimen/margin_extra_large"
-                android:layout_marginTop="@dimen/margin_extra_large"
+                android:layout_width="@dimen/image_major_64"
+                android:layout_height="@dimen/image_major_64"
+                android:layout_marginStart="@dimen/major_100"
+                android:layout_marginTop="@dimen/major_100"
                 android:importantForAccessibility="no"
                 app:layout_constraintStart_toStartOf="parent"
                 app:layout_constraintTop_toBottomOf="@+id/divider"
                 tools:srcCompat="@drawable/ic_user_circle_grey_24dp"/>
```
[the remark this answers is not in the document]
Comment (Garance91540) [reply WooCommerce/src/main/res/layout/fragment_review_detail.xml]: I'd say Status Gray Gray 5 • #DCDCDE • rgb(220,220,222)

File: WooCommerce/src/main/res/layout/fragment_review_detail.xml (hunk 2 above)
Comment (Garance91540) [reply WooCommerce/src/main/res/layout/fragment_review_detail.xml]: I'd say Status Gray Gray 5 • #DCDCDE • rgb(220,220,222) as well!

**Comment by 0nko** [comment]
@kyleaparker Yeah, the dark version uses a shade of orange. The light version uses the color from Figma (Reviews v1), but that would look too dark on a dark background. Do you want me to change it?

**Comment by kyleaparker** [comment]
>@kyleaparker Yeah, the dark version uses a shade of orange. The light version uses the color from Figma (Reviews v1), but that would look too dark on a dark background. Do you want me to change it?

Yep, let's use the Muriel light/dark yellows from here: https://woomobiledesignhandbook.wordpress.com/colors/

File: WooCommerce/src/main/res/layout/fragment_review_detail.xml [hunk 6]
```diff
@@ -12,135 +12,131 @@
...
                 android:id="@+id/review_open_product"
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:layout_marginTop="@dimen/margin_small"
+                android:layout_marginTop="@dimen/minor_50"
                 android:background="?attr/selectableItemBackground"
                 android:gravity="center_vertical"
                 android:orientation="horizontal"
-                android:padding="@dimen/margin_medium"
+                android:padding="@dimen/minor_100"
```
[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/layout/fragment_review_detail.xml]: I didn't add or remove paddings/margins, I was just replacing the existing resource references with the new ones.

[the remark this answers is not in the document]
Comment (0nko) [reply WooCommerce/src/main/res/layout/fragment_review_detail.xml]: The paddings are necessary here for the proper selectable background and margin at the top adds symmetry to the bottom margin (declared by the view below).

File: WooCommerce/src/main/res/values/colors.xml (hunk 4 above)
Comment (0nko) [reply WooCommerce/src/main/res/values/colors.xml]: 🤦‍♂ You're right!

**Comment by 0nko** [comment]
Thanks for the review, @AmandaRiu! I've addressed your feedback comments and the PR's ready for another round of review.

I've updated the product placeholder icon, which affects other screens, such as the product list. I've changed the icon color to On Surface / Medium Emphasis (both product icon and avatar icon) and now a different color is displayed based on the current theme:

image
image

image
image

cc: @kyleaparker @Garance91540

File: WooCommerce/src/main/res/layout/fragment_review_detail.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (AmandaRiu) [reply WooCommerce/src/main/res/layout/fragment_review_detail.xml]: Ah, yup, I see what you mean. 👍

**Pull request merged**